networking_generic_switch/devices/netmiko_devices/__init__.py

```diff
@@ -23,14 +24,53 @@
 import tenacity
 from tooz import coordination
 
+from netmiko.py23_compat import string_types
+
 from networking_generic_switch import devices
 from networking_generic_switch import exceptions as exc
 from networking_generic_switch import locking as ngs_lock
 
+
 LOG = logging.getLogger(__name__)
 CONF = cfg.CONF
 
 
+def send_config_set(net_connect, config_commands=None, exit_config_mode=True,
+                    delay_factor=1, max_loops=150, strip_prompt=False,
+                    strip_command=False, config_mode_cmd=''):
+    """Temporarily overriding the send_config_set method from netmiko.
+
+    Temporarily overriding the send_config_set method from netmiko, until
+    upstream patch is accepted:
+
+    https://github.com/ktbyers/netmiko/pull/593
+
+    """
+
+    delay_factor = net_connect.select_delay_factor(delay_factor)
+    if config_commands is None:
+        return ''
+    elif isinstance(config_commands, string_types):
+        config_commands = (config_commands,)
+
+    if not hasattr(config_commands, '__iter__'):
+        raise ValueError("Invalid argument passed into send_config_set")
+
+    # Send config commands
+    output = net_connect.config_mode(config_mode_cmd)
+    for cmd in config_commands:
+        net_connect.write_channel(net_connect.normalize_cmd(cmd))
+        time.sleep(delay_factor * .5)
+
+    # Gather output
+    output += net_connect._read_channel_timing(
+        delay_factor=delay_factor, max_loops=max_loops)
+    if exit_config_mode:
+        output += net_connect.exit_config_mode()
+    output = net_connect._sanitize_output(output)
+    return output
+
+
```

Comment:
This entire method is lifted from netmiko, with the only alterations being the selfs being changed to net_connects, config_mode now accepting config_mode_cmd as an arg, and the appropriate extra args passed in at the top